README.md

```diff
@@ -498,6 +498,99 @@ Flag | Notes
 `--ftp_proxy`       | Needed for some Perfkit packages
 
+CONFIGURATIONS AND CONFIGURATION OVERRIDES
+==================
+Each benchmark now has an independent configuration which is written in YAML.
+A user config is a way to override this default configuration on a per benchmark
+basis. This allows for much more complex setups than previously possible,
+including running benchmarks across clouds.
+
+A benchmark configuration has a somewhat simple structure. It is essentially
+just a series of nested dictionaries. At the top level, it contains VM groups.
+VM groups are logical groups of homogenous machines. The VM groups hold both a
+"vm_spec" and a "disk_spec" which contain the parameters needed to
+create members of that group. Here is an example of an expanded configuration:
+```
+hbase_ycsb:
+  vm_groups:
+    loaders:
+      vm_count: 4
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-1
+          image: ubuntu-14-04
+          zone: us-central1-c
+        AWS:
+          machine_type: m3.medium
+          image: ami-######
+          zone: us-east-1a
+        # Other clouds here...
+      # This specifies the cloud to use for the group. This allows for
+      # benchmark configurations that span clouds.
+      cloud: AWS
+      # No disk_spec here since these are loaders.
+    master:
+      vm_count: 1
+      cloud: GCP
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-1
+          image: ubuntu-14-04
+          zone: us-central1-c
+        # Other clouds here...
+      disk_count: 1
+      disk_spec:
+        GCP:
+          disk_size: 100
+          disk_type: standard
+          mount_point: /scratch
+        # Other clouds here...
+    workers:
+      vm_count: 4
+      cloud: GCP
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-4
+          image: ubuntu-14-04
+          zone: us-central1-c
+        # Other clouds here...
+      disk_count: 1
+      disk_spec:
+        GCP:
+          disk_size: 500
+          disk_type: remote_ssd
+          mount_point: /scratch
+        # Other clouds here...
+```
+
+For a complete list of keys for vm_specs and disk_specs see
+[virtual_machine.BaseVmSpec](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/virtual_machine.py)
+and
+[disk.BaseDiskSpec](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/disk.py)
+and their derived classes.
+
+User configs are applied on top of the existing default config and can be
+specified in two ways. The first is by supplying a config file via the
+`--benchmark_config_file` flag. The second is by specifying a single setting to
+override via the `--config_override` flag.
+
+A user config file only needs to specify the settings which it is intended to
+override. For example if the only thing you want to do is change the number of
+VMs in a group, this config is sufficient:
```

Comment:
Nit: could you be specific here? "the number of VMs for the cluster_boot benchmark"?